src/client/__init__.py

```diff
@@ -2604,16 +2606,79 @@ def get_operation(self, operation_id: str) -> operations.Operation:
         response = self._request('get', f'/v1/operations/{operation_id}')
         return structure(response, operations.Operation)
 
+    @add_headers('client')
+    def wait_operation_with_progress(
```

Comment:
`tqdm.tqdm` has `disable` parameter in the constructor. Let's not create a separate function and pass the inverted `progress` parameter as `disable` instead. I believe that the usage of `logging_redirect_tqdm` should also not be a problem even in the case of `progress=False`.